lib/logstash/util/compress.rb

```diff
@@ -0,0 +1,109 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/environment"
+require "zip"
+require "rubygems/package"
+require "fileutils"
+require "tempfile"
+require "zlib"
+
+module LogStash::Util
+  module Zip
+
+    extend self
+
+    # Extract a zip file into a destination directory.
+    # @param source [String] The location of the file to extract
+    # @param target [String] Where you do want the file to be extracted
+    def extract(source, target)
+      ::Zip::File.open(source) do |zip_file|
+        zip_file.each do |file|
+          path = File.join(target, file.name)
+          FileUtils.mkdir_p(File.dirname(path))
```

Comment:
I wonder if trying to create the base dir over and over is more efficient than checking if that dir `exits?` first?